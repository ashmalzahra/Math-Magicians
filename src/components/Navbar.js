import { NavLink } from 'react-router-dom';

const links = [
  { path: '/', text: 'Home' },
  { path: 'calculator', text: 'Calculator' },
  { path: 'quote', text: 'Quote' },
];

const Navbar = () => {
  const navStyle = {
    display: 'flex',
    listStyle: 'none',
    justifyContent: 'space-between',
    lineHeight: '50px',
    paddingBottom: '2%',
  };
  const listStyle = {
    paddingRight: '8%',
    fontSize: 'x-large',
  };
  return (
    <nav>
      <ul style={navStyle}>
        <li style={{ fontSize: 'xxx-large' }}>Math Magicians</li>
        <div style={{
          display: 'flex',
          paddingRight: '10%',
        }}
        >
          {links.map((link) => (
            <li key={link.text} style={listStyle}>
              <NavLink
                to={link.path}
                style={({ isActive }) => ({
                  color: isActive ? '#007cff' : '#9e9e9e',
                })}
              >
                {link.text}
              </NavLink>
            </li>
          ))}
        </div>
      </ul>
    </nav>
  );
};

export default Navbar;
