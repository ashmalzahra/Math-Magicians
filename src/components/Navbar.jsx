import { NavLink } from 'react-router-dom';

const links = [
  { path: '/', text: 'Home' },
  { path: 'calculator', text: 'Calculator' },
  { path: 'quote', text: 'Quote' },
];

const Navbar = () => (
  <nav>
    <ul>
      <li><h1>Math Magicians</h1></li>
      {links.map((link) => (
        <li key={link.text}>
          <NavLink to={link.path}>{link.text}</NavLink>
        </li>
      ))}
    </ul>
  </nav>
);

export default Navbar;
