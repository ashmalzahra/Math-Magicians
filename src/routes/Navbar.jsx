import { NavLink } from 'react-router-dom';

const links = [
  { path: '/', text: 'Home' },
  { path: 'calculator', text: 'Calculator' },
  { path: 'quote', text: 'Quote' },
];

const Navbar = () => (
  <nav>
    <ul>
      {links.map((link) => (
        <li key={link.text}>
          <NavLink to={link.path}>{link.text}</NavLink>
        </li>
      ))}
    </ul>
  </nav>
);

export default Navbar;
