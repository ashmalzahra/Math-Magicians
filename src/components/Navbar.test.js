import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Navbar from './Navbar';

describe('Checking page deployment', () => {
  test('Navbar component matches snapshot', () => {
    const tree = renderer.create(<Router><Navbar /></Router>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('renders Navbar component', () => {
    render(<Router><Navbar /></Router>);

    expect(screen.getByRole('navigation')).toBeInTheDocument();
    expect(screen.getByText(/Home/)).toBeInTheDocument();
  });
});
