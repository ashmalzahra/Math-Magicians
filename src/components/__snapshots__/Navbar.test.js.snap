// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking page deployment Navbar component matches snapshot 1`] = `
<nav>
  <ul
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-between",
        "lineHeight": "50px",
        "listStyle": "none",
        "paddingBottom": "2%",
      }
    }
  >
    <li
      style={
        Object {
          "fontSize": "xxx-large",
        }
      }
    >
      Math Magicians
    </li>
    <div
      style={
        Object {
          "display": "flex",
          "paddingRight": "10%",
        }
      }
    >
      <li
        style={
          Object {
            "fontSize": "x-large",
            "paddingRight": "8%",
          }
        }
      >
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
          style={
            Object {
              "color": "#007cff",
            }
          }
        >
          Home
        </a>
      </li>
      <li
        style={
          Object {
            "fontSize": "x-large",
            "paddingRight": "8%",
          }
        }
      >
        <a
          className=""
          href="/calculator"
          onClick={[Function]}
          style={
            Object {
              "color": "#9e9e9e",
            }
          }
        >
          Calculator
        </a>
      </li>
      <li
        style={
          Object {
            "fontSize": "x-large",
            "paddingRight": "8%",
          }
        }
      >
        <a
          className=""
          href="/quote"
          onClick={[Function]}
          style={
            Object {
              "color": "#9e9e9e",
            }
          }
        >
          Quote
        </a>
      </li>
    </div>
  </ul>
</nav>
`;
